h2o-py/h2o/explanation/_explain.py

```diff
@@ -1511,6 +1519,38 @@ def varimp_heatmap(
     >>> aml.varimp_heatmap()
     """
     plt = get_matplotlib_pyplot(False, raise_if_not_available=True)
+    varimps, model_ids,  x = varimp(models=models, top_n=top_n, cluster=cluster, use_pandas=False)
+
+    plt.figure(figsize=figsize)
+    plt.imshow(varimps, cmap=plt.get_cmap(colormap))
+    plt.xticks(range(len(model_ids)), model_ids,
+               rotation=45, rotation_mode="anchor", ha="right")
+    plt.yticks(range(len(x)), x)
+    plt.colorbar()
+    plt.xlabel("Model Id")
+    plt.ylabel("Feature")
+    plt.title("Variable Importance Heatmap")
+    plt.grid(False)
+    fig = plt.gcf()
+    return fig
+
+
+def varimp(
+        models,  # type: Union[h2o.automl._base.H2OAutoMLBaseMixin, List[h2o.model.ModelBase]]
+        top_n=20,  # type: int
+        cluster=True,  # type: bool
+        use_pandas=True  # type: bool
+):
+    # type: (...) -> Union[pandas.DataFrame, Tuple[numpy.ndarray, List[str], List[str]]]
+    """
+        Get data that are used to build varimp_heatmap plot.
+
+        :param models: H2O AutoML object or list of H2O models
+        :param top_n: use just top n models (applies only when used with H2OAutoML)
```

Comment:
are we already using this param with this semantic?
I don't like that:
- it has a default value, although it applies only to AutoML: I would either assign a (non-None) default value and ensure it always applies, or set it to None and assign the default dynamically if AutoML.
- it's in 2nd position although the param doesn't apply when using lists: we should at least ensure that the first 3 params are always used, given that we can't enforce keywords params with the `*` marker (Py 3 only).
- it works only with the default ordering of the AutoML leaderboard.

I think we would have been better advised to provide a `get_top_n_models` method on AutoML.
For example:
```python
# in H2OAutoMLBase
def get_top_models(self, top_n=20, order_by=None):
    pass

# in def varimp(…):
    if isinstance(models, h2o.automl._base.H2OAutoMLBaseMixin):
        models = models.get_top_models()
```
No need for `top_n` param here anymore, and if user wants to customize the models extraction from AutoML, it's easy to do:
```python
varimp(aml.get_top_models(30, order_by='logloss'), cluster=False)
```
without mentioning that it removes duplicate code from all those places where we use `top_n` in this file.

Reply:
The `top_n` is already used in explain.

The problem with `explain` is that it has different behavior depending on the input -  for automl it will show you mainly the leader model and SHAP from the best tree model, for list of models it assumes that you handpicked them and want to see each of the handpicked models. This would make it hard to use the `get_top_models`. I'm not sure if there is some easy intuitive way of doing this differently.

That being said, I'm not opposed to changing the behavior but it would not be backward compatible and I think it is easier for the users to, e.g., in Jupyter press Shift + Tab and find out about `top_n` than going through the methods(though I think most people would just try to get a subset of a list of models from the leaderboard).